### PR TITLE
Troubleshooting guide for diagnosing unassigned shards

### PR DESCRIPTION
**Pull request by andreidan**
This adds a generic guide for diagnosing shards. This guide will be used 
as part of a user action for the shards_availability health indicator when
the API cannot detect a precise cause for the unassigned shards.

### CODE REVIEW

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+image::images/kibana-console.png[{kib} Console,align="center"]
+
+. View the unassigned shards using the <<cat-shards,cat shards API>>.
++
+[source,console]
+----
+GET _cat/shards?v=true&h=index,shard,prirep,state,node,unassigned.reason&s=state
+----
++
+The response will look like this:
```
Comment (Leaf-Lin) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:45]: The following response is not from `GET _cat/shards` by default.  Do you mean to add `&format=json`?

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  "index": "my-index-000001", <1>
+  "shard": 0, <2>
+  "primary": true <3>
+}
+----
+// TEST[skip:illustration purposes only]
++
+<1> The index we want to diagnose.
++
+<2> The unassigned shard.
```
Comment (Leaf-Lin) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:87]:
```suggestion
<2> The unassigned shard ID.
```

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+correct values in order to allow the index to be allocated.
+
+For more guidance on fixing the most common causes for unassinged shards please follow
+<<fix-red-yellow-cluster-status, this guide>> or contact https://support.elastic.co[Elastic Support]. 
+
+//end::kibana-api-ex[]
+// end::cloud[]
+
+// tag::self-managed[]
+In order to get diagnoste the unassigned shards follow the next steps:
```
Comment (Leaf-Lin) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:163]:
```suggestion
In order to diagnose the unassigned shards follow the next steps:
```

Same as above

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+  "index": "my-index-000001", <1>
+  "shard": 0, <2>
+  "primary": true <3>
+}
+----
+// TEST[skip:illustration purposes only]
++
+<1> The index we want to diagnose.
++
+<2> The unassigned shard.
```
Comment (Leaf-Lin) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:214]:
```suggestion
<2> The unassigned shard ID.
```

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
++
+The index in the example has a primary shard unassigned.
+
+. To understand why an unassigned shard is not being assigned and what action
+you must take to allow {es} to assign it, use the
+<<cluster-allocation-explain,cluster allocation explanation API>>.
++
+[source,console]
+----
+GET _cluster/allocation/explain?filter_path=index,node_allocation_decisions.node_name,node_allocation_decisions.deciders.*
```
Comment (Leaf-Lin) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:203]:
```suggestion
GET _cluster/allocation/explain
```

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+// end::cloud[]
+
+// tag::self-managed[]
+In order to get diagnoste the unassigned shards follow the next steps:
+
+. View the unassigned shards using the <<cat-shards,cat shards API>>.
++
+[source,console]
+----
+GET _cat/shards?v=true&h=index,shard,prirep,state,node,unassigned.reason&s=state
```
Comment (Leaf-Lin) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:169]:
```suggestion
GET _cat/shards?v=true&h=index,shard,prirep,state,node,unassigned.reason&s=state&format=json
```

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+If the name of your deployment is disabled your {kib} instances might be
+unhealthy, in which case please contact https://support.elastic.co[Elastic Support],
+or your deployment doesn't include {kib}, in which case all you need to do is 
+{cloud}/ec-access-kibana.html[enable Kibana first].
+
+. Open your deployment's side navigation menu (placed under the Elastic logo in the upper left corner)
+and go to **Dev Tools > Console**.
++
+[role="screenshot"]
+image::images/kibana-console.png[{kib} Console,align="center"]
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:36]: I feel like I've seen this sentence enough that we should make it an expandable token, like we do with `{es}` or `{kib}`

File: docs/reference/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 8]
```diff
@@ -0,0 +1,12 @@
+[[diagnose-unassigned-shards]]
+== Diagnose unassigned shards
+
+There are multiple reasons why shards might get unassigned, ranging from misconfigured
+allocation settings to lack of disk space.
+
+In order to diagnose the unassigned shards in your deployment follow the next steps:
```
Comment (dakrone) [inline docs/reference/troubleshooting/data/diagnose-unassigned-shards.asciidoc:7]: This sounds a little oddly worded, what do you think of this wording?

```suggestion
In order to diagnose the unassigned shards in your deployment use the following steps:
```

**Review by dakrone — APPROVED** [review]
LGTM, I left a few comments

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 9]
```diff
@@ -0,0 +1,287 @@
+////
+[source,console]
+----
+PUT my-index-000001?master_timeout=1s&timeout=1s
```
[the remark this answers is not in the document]
Comment (andreidan) [reply docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:4]: The index will not get assigned - it's designed to remain unassigned at creation time. This shortcircuits the 30 seconds default wait time for the index to be green

[the remark this answers is not in the document]
Comment (dakrone) [reply docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:4]: Ahh of course! Makes sense, thanks

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc (hunk 7 above)
Comment (andreidan) [reply docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:36]: Would it be ok if we do this in a follow-up? (I have no idea how to do this and it'll be easy enough to, once done, replace in these guides)

Comment (dakrone) [reply docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:36]: Oh totally, not a blocker for this at all

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
++
+
+. On the **Elasticsearch Service** panel, click the name of your deployment. 
++
+
+NOTE:
+If the name of your deployment is disabled your {kib} instances might be
+unhealthy, in which case please contact https://support.elastic.co[Elastic Support],
+or your deployment doesn't include {kib}, in which case all you need to do is 
+{cloud}/ec-access-kibana.html[enable Kibana first].
```
Comment (jbaiera) [inline docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:30]:
```suggestion
If the name of your deployment is disabled your {kib} instances might be
unhealthy, in which case please contact https://support.elastic.co[Elastic Support]. 
If your deployment doesn't include {kib}, all you need to do is 
{cloud}/ec-access-kibana.html[enable it first].
```

**Review by jbaiera — APPROVED** [review]
LGTM. One suggestion

File: docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc (hunk 6 above)
Comment (andreidan) [reply docs/reference/tab-widgets/troubleshooting/data/diagnose-unassigned-shards.asciidoc:169]: I'll remove the enclosing { } to not make the request more complex

**Pull request merged**